### PR TITLE
openblas 0.3.13

### PR DESCRIPTION
**Pull request by carlocab**
Created with `brew bump-formula-pr`.

|Formula|CI Failure|Version|Comment|
|---|---|---|---|
|mapnik|`fetch`|10.14,10.15,11.0|https://github.com/Homebrew/homebrew-core/pull/65895|
|networkit|`test`|10.15,11.0|https://github.com/Homebrew/homebrew-core/issues/66450|

### CODE REVIEW

**Comment by chenrui333** [comment]
```
 gcc-ranlib ../../libopenblasp-r0.3.13.a
 make[2]: gcc-ranlib: No such file or directory
 make[2]: *** [../../libopenblasp-r0.3.13.a] Error 1
 make[1]: *** [lapacklib] Error 2
 make: *** [netlib] Error 2
```

**Comment by carlocab** [comment]
Ah. That's because it needs `gcc-ranlib-10`, but is trying to call it as `gcc-ranlib.`

File: Formula/openblas.rb [hunk 1]
```diff
@@ -33,8 +32,11 @@ def install
       Hardware.oldest_cpu.upcase.to_s
     end
 
+    # make looks for `gcc-ranlib`, so we have to point it to the right one
+    ranlib = "RANLIB=gcc-ranlib-#{Formula["gcc"].version.major}"
+
     # Must call in two steps
-    system "make", "CC=#{ENV.cc}", "FC=gfortran", "libs", "netlib", "shared"
+    system "make", "CC=#{ENV.cc}", "FC=gfortran", ranlib, "libs", "netlib", "shared"
```
Comment (fxcoudert) [inline Formula/openblas.rb:39]: How about `RANLIB=ranlib`?

Comment (carlocab) [reply Formula/openblas.rb:39]: Sure, can do. Just thought this was more compact.

Comment (fxcoudert) [reply Formula/openblas.rb]: No, I mean using the system `ranlib`.

File: Formula/openblas.rb [hunk 2]
```diff
@@ -33,8 +32,11 @@ def install
       Hardware.oldest_cpu.upcase.to_s
     end
 
+    # make looks for `gcc-ranlib`, so we have to point it to the right one
+    ranlib = "gcc-ranlib-#{Formula["gcc"].version.major}"
+
```
Comment (fxcoudert) [inline Formula/openblas.rb:37]:
```suggestion
```

File: Formula/openblas.rb [hunk 3]
```diff
@@ -33,8 +32,11 @@ def install
...
     # Must call in two steps
-    system "make", "CC=#{ENV.cc}", "FC=gfortran", "libs", "netlib", "shared"
+    system "make", "CC=#{ENV.cc}", "FC=gfortran", "RANLIB=#{ranlib}", "libs", "netlib", "shared"
```
Comment (fxcoudert) [inline Formula/openblas.rb:39]:
```suggestion
    system "make", "CC=#{ENV.cc}", "FC=gfortran", "RANLIB=ranlib", "libs", "netlib", "shared"
```

File: Formula/openblas.rb (hunk 1 above)
Comment (carlocab) [reply Formula/openblas.rb]: Oh, I see. Ok, let's try that.

**Comment by carlocab** [comment]
10.14:
```
Error: 1 failed steps!
brew fetch --retry mapnik
```
Should hopefully be fixed in https://github.com/Homebrew/homebrew-core/pull/65895

**Comment by fxcoudert** [comment]
Upstream issue: https://github.com/xianyi/OpenBLAS/issues/3037

**Comment by fxcoudert** [comment]
This is actually a real openblas bug, so let's not merge and wait for a proper fix: https://github.com/xianyi/OpenBLAS/issues/3037#issuecomment-744058712

**Comment by carlocab** [comment]
11.0
```
Error: 2 failed steps!
brew fetch --retry mapnik
brew test --retry --verbose networkit
```

10.15
```
Error: 1 failed steps!
brew fetch --retry mapnik
```

**Comment by carlocab** [comment]
10.14:
```
Error: 1 failed steps!
brew fetch --retry mapnik
```
I actually think this is ready to merge. The CI failures are documented elsewhere, and the upstream fix affects the build only. Of course, one can be concerned about bitrot, so I'm willing to wait for a little while too.

**Comment by fxcoudert** [comment]
I'd prefer to use the upstream patch, clearly noting that it can be removed in the next version: https://github.com/xianyi/OpenBLAS/pull/3038

Otherwise we'll forget to remove that `RANLIB` thing, and also there might be other issues by the build thinking it's cross-compiled

**Review by fxcoudert — APPROVED** [review]

**Comment by carlocab** [comment]
I think auto-merge is skipping this?

**Pull request closed without merging**